### PR TITLE
[PWGEM/PhotonMeson] Fix abs in emcalQC task

### PR DESCRIPTION
**Pull request by nstrangm**
- Implemented bugfix for "abs-bug" reported by Vit via the o2linter (https://github.com/AliceO2Group/O2Physics/pull/8942)

### CODE REVIEW

File: PWGEM/PhotonMeson/Tasks/emcalQC.cxx [hunk 1]
```diff
@@ -156,16 +160,16 @@ struct emcalQC {
...
-      if (eventcuts.onlyKeepWeightedEvents && fabs(collision.weight() - 1.) < 1E-10) {
+      if (eventcuts.onlyKeepWeightedEvents && std::fabs(collision.weight() - 1.) < 1E-10) {
         continue;
       }
 
       fRegistry.fill(HIST("Event/hEMCCollisionCounter"), 1);
       if (collision.selection_bit(o2::aod::evsel::kIsTriggerTVX)) {
         fRegistry.fill(HIST("Event/hEMCCollisionCounter"), 2);
-        if (abs(collision.posZ()) < eventcuts.cfgZvtxMax) {
+        if (collision.posZ() < eventcuts.cfgZvtxMax && collision.posZ() > -eventcuts.cfgZvtxMax) {
```
Comment (vkucera) [inline PWGEM/PhotonMeson/Tasks/emcalQC.cxx:172]: Why did you remove the `abs`?

**Review by vkucera — CHANGES_REQUESTED** [review]

File: PWGEM/PhotonMeson/Tasks/emcalQC.cxx (hunk 1 above)
Comment (nstrangm) [reply PWGEM/PhotonMeson/Tasks/emcalQC.cxx:172]: I got a warning about the abs by itself and then checked out how it was done in the PWGEM/Dilepton/Core/EMEventCut.h and wanted to make it more similar to the way it was implemented there

Comment (vkucera) [reply PWGEM/PhotonMeson/Tasks/emcalQC.cxx:172]: But `EMEventCut` has `mMinZvtx` and `mMaxZvtx`. You only have `cfgZvtxMax`.

Comment (nstrangm) [reply PWGEM/PhotonMeson/Tasks/emcalQC.cxx:172]: Yes, however, I don't plan on ever using an asymmetric z vertex cut.
I don't really understand the point of this change request. Do you not like the style of my if statement, or would you prefer EMCal QC to be done in an asymmetric z vertex region, or what is blocking this PR?

Comment (vkucera) [reply PWGEM/PhotonMeson/Tasks/emcalQC.cxx:172]: I understood the subject of this PR was to fix the abs bug. What you did instead is to change the condition to support asymmetric selection without actually using it.

Comment (nstrangm) [reply PWGEM/PhotonMeson/Tasks/emcalQC.cxx:172]: Sorry, maybe I don't fully understand: Does this not implicitly solve the issue?

Comment (vkucera) [reply PWGEM/PhotonMeson/Tasks/emcalQC.cxx:172]: It's not clear to me why you would prefer
```c++
collision.posZ() < eventcuts.cfgZvtxMax && collision.posZ() > -eventcuts.cfgZvtxMax
```
instead of
```c++
std::abs(collision.posZ()) < eventcuts.cfgZvtxMax
```
if don't use and will never use an asymmetric cut.

Comment (nstrangm) [reply PWGEM/PhotonMeson/Tasks/emcalQC.cxx:172]: To be honest, I do not have a preference at all. But I am happy to comply with the request

Comment (vkucera) [reply PWGEM/PhotonMeson/Tasks/emcalQC.cxx]: Thanks

**Review by vkucera — APPROVED** [review]

**Review by mhemmer-cern — APPROVED** [review]
